cmd/livepeer/starter/starter.go

```diff
@@ -1293,7 +1383,10 @@ func StartLivepeer(ctx context.Context, cfg LivepeerConfig) {
 		*cfg.CliAddr = defaultAddr(*cfg.CliAddr, "127.0.0.1", TranscoderCliPort)
 	}
 
-	n.Capabilities = core.NewCapabilitiesWithConstraints(append(transcoderCaps, aiCaps...), core.MandatoryOCapabilities(), constraints)
+	n.Capabilities = core.NewCapabilitiesWithConstraints(append(transcoderCaps, aiCaps...), core.MandatoryOCapabilities(), core.Constraints{}, capabilityConstraints)
```

Comment:
@victorges I have extended the `core.NewCapabilitiesWithConstraints` method created by Yondon to accept general constraints (`core.Constraints{}`) in addition to capability constraints. Although the general constraints are not currently used, I included this field for consistency. However, this is not entirely correct as the `core.Constraints` type has no public members, making it unusable in its current form.

I would like to hear your thoughts on how we should handle this. I noticed you defined a setter and getter for the MinVersion constraint, likely to enhance encapsulation, validation, consistency, and immutability, and to prevent runtime nil errors. We could consider removing the `NewCapabilitiesWithConstraints` method and instead create a setter for the capability constraints. If we want to keep the current `core.NewCapabilitiesWithConstraints` implementation, we should make its fields public so users can utilize it. However, this approach goes against the setter-getter pattern you currently use. What do you think?

Reply:
Implemented in https://github.com/livepeer/go-livepeer/pull/3115. I now took the setter getter route and removed the NewCapabilitiesWithConstraints we can also take the route of making the fields public but I wanted to keep it in line with the rest of the codebase. Please let me know what you prefer 👍🏻.